Lib/fontbakery/profiles/cjk.py

```diff
@@ -0,0 +1,177 @@
+    "Korean Johab": 21
+}
+
+CJK_UNICODE_RANGE_BITS = {
+    'Hangul Jamo': 28,
+    'CJK Symbols And Punctuation': 48,
+    'Hiragana': 49,
+    'Katakana': 50,
+    'Bopomofo': 51,
+    'Hangul Compatibility Jamo': 52,
+    'Enclosed CJK Letters And Months': 54,
+    'CJK Compatibility': 55,
+    'Hangul Syllables': 56,
+    'CJK Unified Ideographs': 59,
+    'CJK Strokes': 61,
+    'Yi Syllables': 83
+}
+
+UNICODE_RANGES = [
+    [0x1100, 0x11FF],  # Hangul Jamo
+    [0x3000, 0x303F],  # CJK Symbols and Punctuation
+    [0x3040, 0x309F],  # Hiragana
+    [0x30A0, 0x30FF],  # Katakana
+    [0x31F0, 0x31FF],  # Katakana Phonetic Extensions
+    [0x3100, 0x312F],  # Bopomofo
+    [0x31A0, 0x31BF],  # Bopomofo Extended (Bopomofo)
+    [0x3130, 0x318F],  # Hangul Compatibility Jamo
+    [0x3200, 0x32FF],  # Enclosed CJK Letters and Months
+    [0x3300, 0x33FF],  # CJK Compatibility
+    [0xAC00, 0xD7AF],  # Hangul Syllables
+    [0x4E00, 0x9FFF],  # CJK Unified Ideographs
+    [0x2E80, 0x2EFF],  # CJK Radicals Supplement (CJK Unified Ideographs)
+    [0x2F00, 0x2FDF],  # Kangxi Radicals (CJK Unified Ideographs)
+    [0x2FF0, 0x2FFF],  # Ideographic Description Characters (CJK Unified Ideographs)
+    [0x3400, 0x4DBF],  # CJK Unified Ideographs Extension A (CJK Unified Ideographs)
+    [0x20000, 0x2A6DF],  # CJK Unified Ideographs Extension B (CJK Unified Ideographs)
+    [0x3190, 0x319F],  # Kanbun (CJK Unified Ideographs)
+    [0x31C0, 0x31EF],  # CJK Strokes
+    [0xF900, 0xFAFF],  # CJK Compatibility Ideographs (CJK Strokes)
+    [0x2F800, 0x2FA1F],  # CJK Compatibility Ideographs Supplement (CJK Strokes)
+    [0xA000, 0xA48F],  # Yi Syllables
+    [0xA490, 0xA4CF],  # Yi Radicals
+]
+
+# =================================
+# Utility functions
+# =================================
+
+
+def get_scaled_uint32_bit(n):
+    """Returns a 32 bit index value for each of the four uint32 that define the OS/2.ulUnicodeRange 1 through 4.
+    The OpenType specification names these with a sequential index from 0 - 127 but they stored across bits in
+    four separate 32 bit unsigned integers.
+    """
+    if n < 32:
+        return n
+    elif n in range(32, 63):
+        return (n - 32)
+    elif n in range(64, 95):
+        return (n - 64)
+    elif n in range(96, 127):
+        return (n - 96)
+    else:
+        return None
+
+
+def is_kth_bit_set(j, k):
+    if j & (1 << k):
+        return True
+    else:
+        return False
+
+
+# =================================
+# Conditions
+# =================================
+
+
+@condition
+def is_cjk_font(ttFont):
+    """Test font object to confirm that it meets our definition of a CJK font file.
+    The definition is met if any of the following conditions are True:
+       1. The font has a CJK code page bit set in the OS/2 table
+       2. The font has a CJK Unicode range bit set in the OS/2 table
+       3. The font has any CJK Unicode code points defined in the cmap table"""
```

Comment:
@JeremieHornus I am using the following definitions to confirm that a font is a "CJK font":

1. The font has a CJK code page bit set in the OS/2 table
2. The font has a CJK Unicode range bit set in the OS/2 table
3. The font has any CJK Unicode code points defined in the cmap table

Do you feel that there are other criteria that should be part of these checks or that the logic here should be different?  This function will serve as a condition that is used to abort all CJK-specific QA checks in the `check-cjk` module if none of the conditions are met.